src/map.cpp

```diff
@@ -323,9 +323,10 @@ struct TimeOrderedMapBlock {
 /*
 	Updates usage timers
 */
-void Map::timerUpdate(float dtime, float unload_timeout, u32 max_loaded_blocks,
+void Map::timerUpdate(float dtime, float unload_timeout, s32 max_loaded_blocks,
 		std::vector<v3s16> *unloaded_blocks)
 {
+	unload_timeout = std::max(unload_timeout, 0.0f);
```

Comment:
there is a call where this is -1:
<pre>
src/map.cpp
464-
465-void Map::unloadUnreferencedBlocks(std::vector<v3s16> *unloaded_blocks)
466-{
467:	timerUpdate(0.0, -1.0, 0, unloaded_blocks);
468-}
469-
470-void Map::deleteSectors(std::vector<v2s16> &sectorList)
</pre>